build-api/templates/dockerfile.tmplt

```diff
@@ -2,11 +2,13 @@ FROM isldocker/build:v5
 
 RUN eval `ssh-agent -s` && \
     ssh-add ~/.ssh/id_rsa && \
-    ssh-keyscan github.com >> /root/.ssh/known_hosts && \
-    git clone -b {{ .BRANCH }} git@github.com:{{ .REPO_OWNER }}/{{ .REPO_NAME }}.git
-
-RUN cd {{ .REPO_NAME }} && git pull origin {{ .BRANCH }}
-RUN cd {{ .REPO_NAME }} && git checkout {{ .HASH }}
+    ssh-keyscan github.com >> /root/.ssh/known_hosts
+{{if .TAG}}    
+RUN git clone -b {{ .TAG }} git@github.com:{{ .REPO_OWNER }}/{{ .REPO_NAME }}.git
+{{else}}
+RUN git clone -b {{ .BRANCH }} git@github.com:{{ .REPO_OWNER }}/{{ .REPO_NAME }}.git
```

Comment:
@oudeismetis So I updated this, I was more explicit with clone of branch and checkout of the hash. Since the clone of has is still fairly new, it might not work everywhere.  We can look into updating later if we feel it is necessary.